src/components/Avatar/Avatar.svelte

```diff
@@ -0,0 +1,92 @@
+<script lang="ts">
+    import type { Snippet } from 'svelte';
+    import type { Action } from 'svelte/action';
+
+    interface AvatarProps {
+        href?: string;
+        action?: Action<HTMLElement, string>;
+        actionParams?: string;
+
+        // Root (figure)
+        base?: string;
+        background?: string;
+        text?: string;
+        fontWeight?: string;
+        width?: string;
+        border?: string;
+        rounded?: string;
+        shadow?: string;
+        classes?: string;
+
+        // Image
+        imageSrc?: string;
+        imageAlt?: string;
+        imageBase?: string;
+        imageRest?: string;
+
+        // Snippets
+        children?: Snippet;
+    }
+
+    let {
+        href = "",
+        action = () => {},
+        actionParams = "",
+
+        // Root (figure)
+        base = "overflow-hidden isolate",
+        background = "bg-surface-400-500-token",
+        width = "w-16",
+        border = "",
+        rounded = "rounded-full",
+        shadow = "",
+        classes = "",
+
+        // Image
+        imageBase = "w-full object-cover",
+        imageSrc = "",
+        imageAlt = "",
+        imageRest = "",
+
+        // Snippets
+        children,
+    } = $props<AvatarProps>();
+</script>
+
+{#snippet figure()}
+    <figure class="{base} {background} {width} {border} {rounded} {shadow} {classes}">
+        {#if children}
```

Comment:
I would flip the conditional here. "If src" rather than "if child". The image will likely be used more often, and the "fallback" should occur when that is not present.